pkgs/development/mobile/androidenv/platform-tools.nix

```diff
@@ -1,26 +1,33 @@
 {stdenv, zlib, fetchurl, unzip}:
 
+with stdenv.lib;
+
 stdenv.mkDerivation rec {
   version = "26.0.2";
   name = "android-platform-tools-r${version}";
-  src = if (stdenv.system == "i686-linux" || stdenv.system == "x86_64-linux")
-    then fetchurl {
-      url = "https://dl.google.com/android/repository/platform-tools_r${version}-linux.zip";
-      sha256 = "0695npvxljbbh8xwfm65k34fcpyfkzvfkssgnp46wkmnq8w5mcb3";
-    }
-    else if stdenv.system == "x86_64-darwin" then fetchurl {
-      url = "https://dl.google.com/android/repository/platform-tools_r${version}-darwin.zip";
-      sha256 = "0gy7apw9pmnnm41z6ywglw5va4ghmny4j57778may4q7ar751l56";
-    }
-    else throw "System ${stdenv.system} not supported!";
+
+  src = fetchurl ({
+    meta = {
+      description = "Android Platform Tools: adb, fastboot, systrace, etc";
+      license = licenses.gpl2;
+      platforms = [ "i686-linux" "x86_64-linux" "x86_64-darwin" ];
+    };
+    url = "unsupported on ${stdenv.system}";
```

Comment:
how is that error displayed? it might be better to add a `throw` keyword to make it more explicit.

Reply:
I don't like this part too. I would prefer for check-meta to give that error. Hence lazy `fetchurl` patch. But its ugly.